library/core/src/main/java/com/google/android/exoplayer2/mediacodec/MediaCodecRenderer.java

```diff
@@ -1170,5 +1170,19 @@ private static boolean codecNeedsMonoChannelCountWorkaround(String name, Format
     return Util.SDK_INT <= 18 && format.channelCount == 1
         && "OMX.MTK.AUDIO.DECODER.MP3".equals(name);
   }
-
+  /**
+   * Returns whether the decoder is known to be non Adaptive.
+   * <p>
+   * If false is returned then we explicitly override codecIsAdaptive,
+   * setting it to false.
+   *
+   * @param name The decoder name.
+   * @param format The input format.
+   * @return False if the device is known  to be non adaptive .
+   */
+  private static boolean codecSupportsAdaptive(String name, Format format) {
```

Comment:
- Can we call this codecNeedsDisableAdaptationWorkaround for consistency with the methods above this one, and to make it clear this is a workaround and what it does.
- The Format arg is unused. Can we remove it (and the corresponding @param in the javadoc)?
- You'll need to get rid of the negating `!` below so that the new method name makes sense. The javadoc will need updating also.